api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/RPCEndpoint.java

```diff
@@ -32,10 +34,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
         ApiRequestPath requestPath = ApiRequestPath.from(request);
         if (requestPath.isValid()) {
             if (log.isDebugEnabled()) {
-                actionPool.printKeys(); 
+                actionPool.printKeys();
             }
             Action action = actionPool.get(requestPath.getActionName());
             OperationData input = new OperationData(request);
+            if (requestPath.isResourceRequest()) {
+                input.add(RESOURCE_ID, new StringData(requestPath.getResourceId()));
```

Comment:
@wwelling can you also please check this https://github.com/vivo-project/Vitro/blob/3718-api-immutable-param-maps/api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/OperationData.java#L45? What should be removed, this or that code? I suppose we don't need both